### PR TITLE
fix(api): refer commit to a node name instead node id for now

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

### CODE REVIEW

File: engine/api/api_routes.go [hunk 1]
```diff
@@ -212,7 +212,7 @@ func (api *API) InitRouter() {
 	r.Handle("/project/{key}/workflows/{permWorkflowName}/runs/{number}/nodes/{nodeRunID}", r.GET(api.getWorkflowNodeRunHandler))
 	r.Handle("/project/{key}/workflows/{permWorkflowName}/runs/{number}/nodes/{nodeRunID}/stop", r.POST(api.stopWorkflowNodeRunHandler))
 	r.Handle("/project/{key}/workflows/{permWorkflowName}/runs/{number}/nodes/{nodeID}/history", r.GET(api.getWorkflowNodeRunHistoryHandler))
-	r.Handle("/project/{key}/workflows/{permWorkflowName}/runs/{number}/{nodeID}/commits", r.GET(api.getWorkflowCommitsHandler))
+	r.Handle("/project/{key}/workflows/{permWorkflowName}/runs/{number}/{nodeName}/commits", r.GET(api.getWorkflowCommitsHandler))
```
Comment (fsamin) [inline engine/api/api_routes.go]: is there any tests on this route ?

File: engine/api/workflow/dao_node.go [hunk 2]
```diff
@@ -290,22 +318,22 @@ func LoadNodeContext(db gorp.SqlExecutor, store cache.Store, nodeID int64) (*sdk
 	//Unmarshal payload
 	if sqlContext.DefaultPayload.Valid {
```
Comment (fsamin) [inline engine/api/workflow/dao_node.go]: use gorpmapping.JSONNullString

File: engine/api/workflow/dao_node.go [hunk 3]
```diff
@@ -290,22 +318,22 @@ func LoadNodeContext(db gorp.SqlExecutor, store cache.Store, nodeID int64) (*sdk
...
 		if err := json.Unmarshal([]byte(sqlContext.DefaultPayload.String), &ctx.DefaultPayload); err != nil {
-			return nil, sdk.WrapError(err, "LoadNodeContext> Unable to unmarshall context %d default payload", ctx.ID)
+			return sdk.WrapError(err, "LoadNodeContext> Unable to unmarshall context %d default payload", ctx.ID)
 		}
 	}
 
 	//Unmarshal pipeline parameters
 	if sqlContext.DefaultPipelineParameters.Valid {
```
Comment (fsamin) [inline engine/api/workflow/dao_node.go]: use gorpmapping.JSONNullString

**Comment by bnjjj** [comment]
Ported to https://github.com/ovh/cds/pull/1805 with cr fix, tests are coming

**Pull request closed without merging**